File: PrintDiamond/printdiamond.py
#!/usr/bin/env python

print "Enter the diamond width, which must be a positive odd integer:",

size = int(raw_input())

# shouldnt really be nessisary as usually you are guarenteed valid input
if (size % 2) == 0:
    print "Error: Number must be an odd number"
    exit(1)

for count in xrange(1, size+1):
    print " " * (size-count),
    print "* " * count

for count in xrange(1, size):
    print " " * count,
    print "* " * (size-count)

File: PrintDiamond/printdiamond.cpp
#include <iostream>

using namespace std;

int main() {

    cout << "Enter the diamond width, which must be a positive odd integer: ";

    int size;
    cin >> size;

    // shouldnt really be nessisary as usually you are guarenteed valid input
    if ((size % 2) == 0) {
        printf("Error: Number must be an odd number\n");
        return 1;
    }

    int count, index;
    for (count = 1; count < (size+1); count++) {
        for (index = 0; index < (size-count); index++) {
            cout << " ";
        }
        for (index = 0; index < count; index++) {
            cout << "* ";
        }
        cout << endl;
    }

    for (count = 1; count < size; count++) {
        for (index = 0; index < count; index++) {
            cout << " ";
        }
        for (index = 0; index < (size-count); index++) {
            cout << "* ";
        }
        cout << endl;
    }

    return 0;
}

